### PR TITLE
Cleanup Makefile and remove obsolete targets

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What type of PR is this?**

**What this PR does / why we need it**: Cleanup some make targets that were not being used. Image pushing is all done through prow jobs so we don't need the release image make targets.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -101,7 +101,7 @@ KUBE_APISERVER=$(TOOLS_BIN_DIR)/kube-apiserver
 ETCD=$(TOOLS_BIN_DIR)/etcd
 
 # Define Docker related variables. Releases should modify and double check these vars.
-REGISTRY ?= gcr.io/$(shell gcloud config get-value project)
+REGISTRY ?= registry.local/fake
```
Comment (mboersma) [inline Makefile:104]: I assume this is just a placeholder, so we don't try to shell out to `gcloud`?

**Review by mboersma — APPROVED** [review]
/lgtm

File: Makefile (hunk 1 above)
Comment (CecileRobertMichon) [reply Makefile:104]: correct

**Review by devigned — APPROVED** [review]
/lgtm
/approve

**Pull request merged**